lib/cudnn/upsampling.jl

```diff
@@ -0,0 +1,142 @@
+# An implementation for GPU based bilinear upsampling including its gradient
+# The code is a translation from the following files:
+# https://github.com/pytorch/pytorch/blob/master/caffe2/operators/upsample_op.cu
+# https://github.com/pytorch/pytorch/blob/master/caffe2/core/common_gpu.h
+
+# Forward and backward pass have been tested to produce the same output
+# as pytorch - it works modulo bit noise.
+
+const CUDA_NUM_THREADS = 128
+const MAXIMUM_NUM_BLOCKS = 4096
+
+@inline function GET_BLOCKS(N::Integer)
+    # Use at least 1 block, since CUDA does not allow empty block
+    return max(min((N + CUDA_NUM_THREADS - 1) ÷ CUDA_NUM_THREADS, MAXIMUM_NUM_BLOCKS), 1)
+end
+
+# pytorch: nchw with row major
+# flux: whcn with column major -> same data layout in memory -> this function
+#   can stay as it is except for +1
+@inline function idx(
```

Comment:
I'd rather we use regular ND-arrays than this style of linear index calculation.